contribs/drt-extensions/src/main/java/org/matsim/contrib/drt/extension/edrt/scheduler/EmptyVehicleChargingScheduler.java

```diff
@@ -52,14 +56,16 @@ public EmptyVehicleChargingScheduler(MobsimTimer timer, DrtTaskFactory taskFacto
 		linkToChargerMap = chargingInfrastructure.getChargers()
 				.values()
 				.stream()
-				.collect(Collectors.toMap(c -> c.getLink().getId(), c -> c));
+				.collect(Collectors.groupingBy(c -> c.getLink().getId()));
 	}
 
 	public void chargeVehicle(DvrpVehicle vehicle) {
 		DrtStayTask currentTask = (DrtStayTask)vehicle.getSchedule().getCurrentTask();
 		Link currentLink = currentTask.getLink();
-		Charger charger = linkToChargerMap.get(currentLink.getId());
-		if (charger != null) {
+		List<Charger> chargers = linkToChargerMap.get(currentLink.getId());
+		if (chargers != null) {
+			// Pick a random charger of all available chargers at this link
+			Charger charger = chargers.get(RND.nextInt(chargers.size()));
```

Comment:
Maybe instead take the one with highest number of unused plugs. If there is none with unused plugs, take the one with shortest queue or lowest `queue.size/plugCount`. You could also include charger power in choosing the charger.

Reply:
Good idea! Thanks.